homeassistant/helpers/service.py

```diff
@@ -630,12 +636,34 @@ async def async_get_all_descriptions(
 
                 # Don't warn for missing services, because it triggers false
                 # positives for things like scripts, that register as a service
+                #
+                # When name & description are in the translations use those;
+                # otherwise fallback to backwards compatible behavior from
+                # the time when we didn't have translations for descriptions yet.
+                # This mimics the behavior of the frontend.
                 description = {
-                    "name": yaml_description.get("name", ""),
-                    "description": yaml_description.get("description", ""),
+                    "name": translations.get(
+                        f"component.{domain}.services.{service_name}.name",
+                        yaml_description.get("name", ""),
+                    ),
+                    "description": translations.get(
+                        f"component.{domain}.services.{service_name}.description",
+                        yaml_description.get("description", ""),
+                    ),
                     "fields": yaml_description.get("fields", {}),
                 }
 
+                # Translate fields names & descriptions as well
+                for field_name, field_schema in description.get("fields", {}).items():
+                    if name := translations.get(
+                        f"component.{domain}.services.{service_name}.fields.{field_name}.name"
+                    ):
+                        field_schema["name"] = name
```

Comment:
You are mutating the source YAML data as `fields` is a reference to the value from `yaml_description`. Now it should only generate this once per service so it _should_ be fine, but just wanted to call it out.